.github/workflows/macos-build.yaml

```diff
@@ -0,0 +1,171 @@
+name: MacOS build
+
+on:
+  push:
+    branches:
+     - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  macos-staging:
+    runs-on: macos-latest
+
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v2
+
+      - name: Checkout submodules
+        shell: bash
+        run: |
+          auth_header="$(git config --local --get http.https://github.com/.extraheader)"
+          git submodule sync --recursive
+          git -c "http.extraheader=$auth_header" -c protocol.version=2 submodule update --init --force --recursive --depth=1
+
+      - name: Install Qt
+        shell: bash
+        run: |
+          auth_header="$(git config --local --get http.https://github.com/.extraheader)"
+          git clone https://github.com/bakulf/qt_static_macos
```

Comment:
is there a reason we're pulling this from a your repo instead of an authoritative source?

Reply:
Because we need a static version of the QT compiled. So far, nobody has exposed the right build.

Reply:
There isn't a tarball we can pull from somewhere? Or are those binaries in your repo?